Sprint 25(home work) Gr26

amanzhola opened the pull request.

amanzhola committed:
Sprint 24(home work) Gr26

[Ревью] Лучшее замечание (!): Создание Bitmap в d2b() может быть дорогой операцией. Лучше кэшировать Bitmap или использовать Drawable напрямую

updates for widgets.kt:
PlaybackButtonView 2 опции решения ->
1) через использование Drawable напрямую
2) через кэширование Bitmap

amanzhola committed:
Sprint 24(home work) Gr26

Updating:
fixing and update toolbar background color for apply and save to activity, fragments, (fragments) children and independent fragments
//*************************
2 new interfaces and 1 new file: ActiveChildProvider, BackgroundExclusionProvider and ScreenKey.kt
//*************************
1. AppModule: ColorPersistenceHelper
2. BaseActivity: + functions and update initialize colorPersistenceHelper and colorManager
3. BaseFragment: +getBaseActivity()?.applySavedColorsForCurrentScreen()
4. ColorApplierHelper: updated segment 1 and include excludeIds for recursion
5. ColorHelper: update saveColor, loadColor, clearColors and generateKey for having on top of activity also fragments
6. ColorManager: increase range to 6
7. ColorPersistenceHelper: update save, load, clear and constructor
8. CreatePlaylistFragment: toolbar background color for independent fragments
9. ExtraOptionFragment: onResume for (activity as? BaseActivity)?.applyThemeThenRestoreSaved()
10. FragmentFavouriteTracks same as above
11. FragmentPlaylist same as above
12. MainFragment: update same as above + exclude
13. MediaLibraryFragment: update with having children separation for toolbar background color apply
14. ScreenKey for having activity and fragment key as names for saving and apply
15. SearchFragment: same as above + exclude for input field
16. SearchMovie same as above + initialization
17. SearchWeather same as above
18. SegmentManager: if (segmentIndex != 5) + toolbar background color apply and save
19. SettingsFragment same as above
20. ToolbarHelper: + fun setBackgroundColor(@ColorInt color: Int)

amanzhola committed:
Sprint 24(home work) Gr26

Re-factoring ExtraOption -> Chat:
1) 2 option with and without tail-> bg_bubble_in.xml, bg_bubble_in_tail.xml, bg_bubble_out.xml, bg_bubble_out_tail.xml,
tail_left_in.xml, tail_right_out.xml, colors.xml, dimens.xml(chat bubble max)
2) build.gradle.kts and Build_gradle (app), libs.versions.toml -> for firebase adds
3) re-name ExtraOptionFragment.kt to AudioPlayerFragment (+ can be taken off logic for entering via btm navigation since ExtraOption finally got ChatFragment)
4) fragment_chat.xml -> new
5) fragment_extra_option.xml and baseline_chat_24.xml -> re-name AudioPlayerFragment + new icon
6) fragment_profile.xml -> new
7) fragment_users.xml -> new
8) google-services.json -> firebase requirement
9) ic_user_placeholder.xml for item on UsersFragment
10) item_message_in.xml, item_message_out.xml and item_user.xml -> message exchange and user ui
11) MainActivity -> launching registration on firebase
12) Message.kt and UsersUi.kz models
13) MessageAdapter -> new
14) NavigationConfigProvider, ScreenType, SegmentManager, SegmentManagerProvider -> by AudioPlayerFragment re-naming from ExtraOptionFragment.kt
15) strings.xml and en add new string resources
16) themes.xml + night for ChatFragment new background color
17) UsersAdapter, UsersFragment.kt, UsersRepo -> new

amanzhola committed:
Update google-services.json

amanzhola committed:
Update google-services.json

amanzhola committed:
Update README.md

Update by 250925

amanzhola committed:
Update README.md

amanzhola committed:
Update README.md

amanzhola committed:
Update google-services.json

Rotate Firebase API key and restrict to Android package+SHA1

amanzhola committed:
Sprint 24(home work) Gr26

Update Chat:
1) gitingore - proper filing
2) AndroidManifest.xml -> Service FCM and Icon for status-bar
3) App.kt -> + 2 notification channels
4) badge_dot_red.xml -> not used: possible needs currently emoji used on ui unread messages
5) BlocksRepo -> blocking and soft deleting
6) Build_gradle -> firebase require files
7) ChatFragment -> resume to make read/unread count (future to add progress bar at loading pictures)
8) ChatItem -> update on read/unread bool
9) ChatMigrations.kt -> try to fix confusions on first starting firebase
10) ChatPushService -> show unread on application icon
11) ChatsRepo -> have logic of sorting in client instead server
12) dialog_new_chat_search.xml - ui on search users
13) FcmTokenManager -> tokens management
14) fragment_users.xml -> add Floating Action Bar(FAB) for search new participants with 2 options: click by name and long click by id
15) google-services.json -> keys security update
16) ic_stat_chat.xml -> ui to show digit or dot for read/unread on application's icon
17) item_chat.xml -> add showing unread via emoji
18) item_message_out.xml and item_message_in.xml -> no updates
19) libs.versions.toml -> cleaning
20) MainActivity -> update on leaving application to counting issue
21) MigrationUtil.kt -> deleted (useless try to refactor this way)
22) playlist_navigation_graph.xml -> adding new fragments
23) SnackbarExt.kt -> add parameter for FAB
24) strings.xml + en -> adds string resources
25) ToolbarHelper -> add fun on UserFragment to ChatFragment for user name at toolbar title
26) UserFragment -> resume update to zero on counting unread messages
27) UsersRepo -> no updates

amanzhola committed:
Sprint 24(home work) Gr26

Update Chat:
1) gitingore - proper filing
2) AndroidManifest.xml -> Service FCM and Icon for status-bar
3) App.kt -> + 2 notification channels
4) badge_dot_red.xml -> not used: possible needs currently emoji used on ui unread messages
5) BlocksRepo -> blocking and soft deleting
6) Build_gradle -> firebase require files
7) ChatFragment -> resume to make read/unread count (future to add progress bar at loading pictures)
8) ChatItem -> update on read/unread bool
9) ChatMigrations.kt -> try to fix confusions on first starting firebase
10) ChatPushService -> show unread on application icon
11) ChatsRepo -> have logic of sorting in client instead server
12) dialog_new_chat_search.xml - ui on search users
13) FcmTokenManager -> tokens management
14) fragment_users.xml -> add Floating Action Bar(FAB) for search new participants with 2 options: click by name and long click by id
15) google-services.json -> keys security update
16) ic_stat_chat.xml -> ui to show digit or dot for read/unread on application's icon
17) item_chat.xml -> add showing unread via emoji
18) item_message_out.xml and item_message_in.xml -> no updates
19) libs.versions.toml -> cleaning
20) MainActivity -> update on leaving application to counting issue
21) MigrationUtil.kt -> deleted (useless try to refactor this way)
22) playlist_navigation_graph.xml -> adding new fragments
23) SnackbarExt.kt -> add parameter for FAB
24) strings.xml + en -> adds string resources
25) ToolbarHelper -> add fun on UserFragment to ChatFragment for user name at toolbar title
26) UserFragment -> resume update to zero on counting unread messages
27) UsersRepo -> no updates

amanzhola committed:
Sprint 25(home work) Gr26

BroadcastReceiver:
1) AudioPlayerFragment -> add var cm, netwrokCallback, lastConnected for ConnectivityManager.NetworkCallBack instead BroadcastReceiver cause it's depreciated CONNECTIVITY_ACTION
+ on resume and pause: since receiver registered between onResume and onPause notification will not appear once application in the background and minimized
2) SearchFragment -> add var cm, netwrokCallback, lastConnected for ConnectivityManager.NetworkCallBack instead BroadcastReceiver cause it's depreciated CONNECTIVITY_ACTION
+ on resume and pause: since receiver registered between onResume and onPause notification will not appear once application in the background and minimized
+ showLongSnack parameter for anchor to bottom navigation height
3) ConnectivityChangeReceiver -> new to presentation package
4) NetworkStatusCheckerImpl -> updated flag from NET_CAPABILITY_INTERNET to NET_CAPABILITY_VALIDATED
5) strings.xml + en -> add string resource for showLongSnack
6) MainActivity -> commented out [if (!wasProfileAskedOnce())] for user's action on filling up profile

amanzhola committed:
Update README.md

update -> + chat(horizontal, unreadCount, vertical)

amanzhola committed:
Sprint 25(home work) Gr26

Service (Bound + Foreground)
1) AndroidManifest.xml -> permissions and MusicService
2) AudioPlayerControl -> interface and PlayerUiState
3) AudioPlayerFragment -> mark as [ADDED] new Permission, AudioPlayerControl, remove viewModel.startObservingAudioPlayer(), onStart, onStop and onDestroyView
4) AudioPlayerInteraction -> expansion of PlaybackState by COMPLETED, ERROR
5) ExtraOptionViewModel -> removing AudioPlayerInteraction and replace by  AudioPlayerControl, managing notifications by went and came to Foreground
6) extraOptionViewModelModule -> Koin update by removing AudioPlayerInteraction from ExtraOptionViewModel
7) MusicService -> media player function
8) strings.xml + en -> string resources
9) TrackPreviewViewModel -> adding PlaybackState by COMPLETED, ERROR
10) extraOptionInteractionModule -> no removal single<AudioPlayerInteraction> cause use of TrackPreviewViewModel notified in comments for future possible transfer for Service (Bound + Foreground)
11) Constants.kt -> for MusicService, AudioPlayerFragment for service Constants.kt

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/services/MusicService.kt:
```diff
@@ -0,0 +1,365 @@
+            _ui.value = PlayerUiState(
+                isButtonEnabled = false,
+                isPlaying = false,
+                progress = "00:00",
```
Лучше не хардкодить этот текст, так как форматирование может легко поменяться и потом придётся по всему проекту искать эти захардкоженные строчки. Можно либо вынести в strings.xml, чтобы это было только в одном месте, либо использовать ту же функцию форматирования через `SimpleDateFormat`, передавая туда значение 0, чтобы поведение было консистентным

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/services/MusicService.kt:
```diff
@@ -0,0 +1,365 @@
+            mediaPlayer?.apply {
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+                    setAudioAttributes(
+                        android.media.AudioAttributes.Builder()
```
Путь к классу лучше заменить на корректный импорт

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/services/MusicService.kt:
```diff
@@ -0,0 +1,365 @@
+        stopTimer()
+        timerJob = CoroutineScope(Dispatchers.Default).launch {
+            while (isActive && mediaPlayer?.isPlaying == true) {
+                delay(200L)
```
Значение задержки лучше вынести в константу для повышения читабельности кода

amanzhola committed:
Sprint 25(home work) Gr26

Review fix update:
1) Constants.kt -> TIMER_INTERVAL_MS, BUTTON_TEXT_PLAY
2) MusicService -> startTimer (const), BUTTON_TEXT_PLAY, getString(R.string._00_00)
import android.media.AudioAttributes for setAudioAttributes
3) strings.xml -> R.string._00_00
